src/widget/wnumberrate.cpp

```diff
@@ -28,16 +28,19 @@ WNumberRate::WNumberRate(const char * group, QWidget * parent)
     setValue(0);
 }
 
-void WNumberRate::setValue(double /*dValue*/) {
-    double vsign = m_pRateControl->get() *
-            m_pRateRangeControl->get() *
-            m_pRateDirControl->get();
+void WNumberRate::setValue(double dValue) {
+    double digitFactor = pow(10, m_iNoDigits);
+    // Calculate percentage rounded to the number of digits specified by iNoDigits
+    double percentage = round((dValue - 1) * 100.0 * digitFactor) / digitFactor;
 
-    char sign = '+';
-    if (vsign < -0.00000001) {
+    QString sign(' ');
+    if (percentage > 0) {
+        sign = '+';
+    }
+    if (percentage < 0) {
         sign = '-';
     }
 
     setText(QString(m_skinText).append(sign)
-            .append("%1").arg(fabs(vsign) * 100.0, 0, 'f', m_iNoDigits));
+            .append("%1").arg(fabs(percentage), 'f', m_iNoDigits);
```

Comment:
```
/home/jan/Projects/mixxx/src/widget/wnumberrate.cpp: In member function ‘virtual void WNumberRate::setValue(double)’:
/home/jan/Projects/mixxx/src/widget/wnumberrate.cpp:46:65: error: call of overloaded ‘arg(double, char, int&)’ is ambiguous
   46 |             .append("%1").arg(fabs(percentage), 'f', m_iNoDigits);
      |                                                                 ^
In file included from /usr/include/qt/QtCore/qobject.h:47,
                 from /usr/include/qt/QtWidgets/qwidget.h:45,
                 from /usr/include/qt/QtWidgets/qframe.h:44,
                 from /usr/include/qt/QtWidgets/qlabel.h:44,
                 from /usr/include/qt/QtWidgets/QLabel:1,
                 from /home/jan/Projects/mixxx/src/widget/wnumber.h:21,
                 from /home/jan/Projects/mixxx/src/widget/wnumberrate.h:15,
                 from /home/jan/Projects/mixxx/src/widget/wnumberrate.cpp:13:
/usr/include/qt/QtCore/qstring.h:313:31: note: candidate: ‘QString QString::arg(qlonglong, int, int, QChar) const’
  313 |     Q_REQUIRED_RESULT QString arg(qlonglong a, int fieldwidth=0, int base=10,
      |                               ^~~
/usr/include/qt/QtCore/qstring.h:315:31: note: candidate: ‘QString QString::arg(qulonglong, int, int, QChar) const’
  315 |     Q_REQUIRED_RESULT QString arg(qulonglong a, int fieldwidth=0, int base=10,
      |                               ^~~
/usr/include/qt/QtCore/qstring.h:1110:16: note: candidate: ‘QString QString::arg(long int, int, int, QChar) const’
 1110 | inline QString QString::arg(long a, int fieldWidth, int base, QChar fillChar) const
      |                ^~~~~~~
/usr/include/qt/QtCore/qstring.h:1112:16: note: candidate: ‘QString QString::arg(ulong, int, int, QChar) const’
 1112 | inline QString QString::arg(ulong a, int fieldWidth, int base, QChar fillChar) const
      |                ^~~~~~~
/usr/include/qt/QtCore/qstring.h:1106:16: note: candidate: ‘QString QString::arg(int, int, int, QChar) const’
 1106 | inline QString QString::arg(int a, int fieldWidth, int base, QChar fillChar) const
      |                ^~~~~~~
/usr/include/qt/QtCore/qstring.h:1108:16: note: candidate: ‘QString QString::arg(uint, int, int, QChar) const’
 1108 | inline QString QString::arg(uint a, int fieldWidth, int base, QChar fillChar) const
      |                ^~~~~~~
/usr/include/qt/QtCore/qstring.h:1114:16: note: candidate: ‘QString QString::arg(short int, int, int, QChar) const’
 1114 | inline QString QString::arg(short a, int fieldWidth, int base, QChar fillChar) const
      |                ^~~~~~~
/usr/include/qt/QtCore/qstring.h:1116:16: note: candidate: ‘QString QString::arg(ushort, int, int, QChar) const’
 1116 | inline QString QString::arg(ushort a, int fieldWidth, int base, QChar fillChar) const
      |                ^~~~~~~
/usr/include/qt/QtCore/qstring.h:329:31: note: candidate: ‘QString QString::arg(double, int, char, int, QChar) const’
  329 |     Q_REQUIRED_RESULT QString arg(double a, int fieldWidth = 0, char fmt = 'g', int prec = -1,
      |                               ^~~
/usr/include/qt/QtCore/qstring.h:331:31: note: candidate: ‘QString QString::arg(char, int, QChar) const’
  331 |     Q_REQUIRED_RESULT QString arg(char a, int fieldWidth = 0,
      |                               ^~~
/usr/include/qt/QtCore/qstring.h:333:31: note: candidate: ‘QString QString::arg(QChar, int, QChar) const’
  333 |     Q_REQUIRED_RESULT QString arg(QChar a, int fieldWidth = 0,
      |                               ^~~
```

Why not something like:
```cpp
    setText(m_skinText + sign + QString::number(fabs(percentage), 'f', m_iNoDigits));
```

Reply:
I haven't tested it, but we could use it

Reply:
Nice. When you fix this and add the function we can merge.